cli/src/git_util.rs

```diff
@@ -172,17 +176,140 @@ pub fn with_remote_git_callbacks<T>(
+            .collect_vec();
+
+        let has_both_ref_kinds = refs_stats
+            .iter()
+            .any(|x| matches!(x.ref_kind, RefKind::Branch))
+            && refs_stats
+                .iter()
+                .any(|x| matches!(x.ref_kind, RefKind::Tag));
+
+        let max_width = refs_stats.iter().map(|x| x.ref_name.width()).max();
+        if let Some(max_width) = max_width {
+            let mut stderr = ui.stderr_formatter();
+            for status in refs_stats {
+                status.output(max_width, has_both_ref_kinds, &mut *stderr)?;
+            }
+        }
+    }
+
     if !stats.abandoned_commits.is_empty() {
         writeln!(
             ui.stderr(),
             "Abandoned {} commits that are no longer reachable.",
             stats.abandoned_commits.len()
         )?;
     }
+
     Ok(())
 }
 
+struct RefStatus {
+    ref_kind: RefKind,
+    ref_name: String,
+    tracking_status: TrackingStatus,
+    import_status: ImportStatus,
+}
+
+impl RefStatus {
+    fn new(
+        ref_name: &RefName,
+        remote_ref: &RemoteRef,
+        ref_target: &RefTarget,
+        repo: &dyn Repo,
+    ) -> Self {
+        let mut tracking_status = TrackingStatus::Untracked;
+        let (ref_name, ref_kind) = match ref_name {
+            RefName::RemoteBranch { branch, remote } => {
+                if repo.view().get_remote_branch(branch, remote).is_tracking() {
+                    tracking_status = TrackingStatus::Tracked;
+                }
+                (format!("{branch}@{remote}"), RefKind::Branch)
+            }
+            RefName::Tag(tag) => {
+                tracking_status = TrackingStatus::NotApplicable;
+                (tag.clone(), RefKind::Tag)
+            }
+            RefName::LocalBranch(branch) => (branch.clone(), RefKind::Branch),
+        };
+
+        let import_status = match (remote_ref.target.is_absent(), ref_target.is_absent()) {
+            (true, false) => ImportStatus::New,
+            (false, true) => ImportStatus::Deleted,
+            _ => ImportStatus::Updated,
+        };
+
+        Self {
+            ref_name,
+            tracking_status,
+            import_status,
+            ref_kind,
+        }
+    }
+
+    fn output(
+        &self,
+        max_ref_name_width: usize,
+        has_both_ref_kinds: bool,
+        out: &mut dyn Formatter,
+    ) -> std::io::Result<()> {
+        let tracking_status = match self.tracking_status {
+            TrackingStatus::Tracked => "tracked",
+            TrackingStatus::Untracked => "untracked",
+            TrackingStatus::NotApplicable => "",
+        };
+
+        let import_status = match self.import_status {
+            ImportStatus::New => "new",
+            ImportStatus::Deleted => "deleted",
+            ImportStatus::Updated => "updated",
+        };
+
+        // This ensures alignment of the statuses that follow
+        let padded_ref_name = format!("{:width$}", self.ref_name, width = max_ref_name_width);
```

Comment:
we'll have to do something like `"{ref_name}{pad:pad_width}", pad = "", pad_width = saturating_sub(max_ref_name_width, ref_name.width())` (untested).

Otherwise the width of `self.ref_name` would be calculated by using `.len()`.